### PR TITLE
KFSPTS-7808: Added CU changes to Current Acct Balance lookup

### PR DESCRIPTION
**Pull request by cah292**
We had numerous customizations to the Current Account/Funds Balance lookup in KFS 5.x, and those needed to be added back in for the uprgade. Because the lookup logic requiring alteration was in a large method on the superclass, the resulting subclass had to just copy the superclass method contents and tweak them accordingly.

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+     * This override copies the superclass's version of the method, and tweaks it to add the CB exclusion option
+     * and to use object type code for Current Asset/Liability checking.
+     * It also fixes a comparison bug in the Current Budget calculation that was getting the right value from the wrong constant.
+     * 
+     * @see org.kuali.kfs.gl.businessobject.lookup.CurrentAccountBalanceLookupableHelperServiceImpl#updateCurrentBalance(
+     * org.kuali.kfs.gl.businessobject.CurrentAccountBalance, org.kuali.kfs.gl.businessobject.Balance, java.lang.String)
+     */
+    @SuppressWarnings("deprecation")
+    @Override
+    protected void updateCurrentBalance(CurrentAccountBalance currentBalance, Balance balance, String fiscalPeriod) {
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: This is a really long method, can we refactor this to make smaller functions that will be easier to ready

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        Collection<String> cashBudgetRecordLevelCodes = this.getParameterService().getParameterValuesAsString(
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: can we drop the "this" from here? and others below

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java (hunk 1 above)
Comment (cah292) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: Ok, I will see if I can split this method up.

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java (hunk 2 above)
Comment (cah292) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: Good catch, I'll remove the unnecessary "this" references.

**Comment by cah292** [comment]
I've made some changes based on initial feedback. This is now ready for review again.

**Comment by cah292** [comment]
retest this please

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+     */
+    @Override
+    protected void updateCurrentBalance(CurrentAccountBalance currentBalance, Balance balance, String fiscalPeriod) {
+        new BalanceUpdater(currentBalance, balance, fiscalPeriod)
+                .updateCurrentBalance();
+    }
+
+    /**
+     * Helper class for encapsulating and splitting up the logic
+     * of the updateCurrentBalance() method.
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: Are these comments needed?  Maybe rename the class to BalanceUpdaterHelper to reinforce that is just a helper class/

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+            this.currentLiabilityObjTypeCodes = getParameterValuesAsString(CUKFSParameterKeyConstants.GlParameterConstants.CURRENT_LIABILITY_OBJECT_TYPE_CODE);
+            this.incomeObjTypeCodes = getParameterValuesAsString(KFSParameterKeyConstants.GlParameterConstants.INCOME_OBJECT_TYPE);
+            this.encumbranceBalTypes = getParameterValuesAsString(KFSParameterKeyConstants.GlParameterConstants.ENCUMBRANCE_BALANCE_TYPE);
+            this.excludeCBPeriod = getParameterValueAsBoolean(CUKFSParameterKeyConstants.GlParameterConstants.EXCLUDE_CB_PERIOD, Boolean.FALSE)
+                    .booleanValue();
+            
+            SystemOptions options = getOptionsService().getCurrentYearOptions();
+            this.assetLiabilityFundBalanceTypeCodes = Arrays.asList(options.getFinancialObjectTypeAssetsCd(),  // AS
+                options.getFinObjectTypeLiabilitiesCode(), // LI
+                options.getFinObjectTypeFundBalanceCd());  // FB
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: I think a few other extraneous comments

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+            updateTotalIncome();
+            updateTotalExpense();
+            updateEncumbrances();
+            updateBudgetBalanceAvailable();
+            updateCashExpenditureAuthority();
+            updateCurrentFundBalance();
+        }
+        
+        /*
+         * NOTE: Base KFS grabs the "CB" balance type from the wrong constant. This has been corrected below.
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: Do we have a contribution Jira to note we should contribute this fix?

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java (hunk 3 above)
Comment (cah292) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: Ok, I will clean up some of the comments and the helper class's name.

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java (hunk 4 above)
Comment (cah292) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: Those were in the base KFS code, but I'll remove them for the sake of cleanliness.

File: src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java (hunk 5 above)
Comment (cah292) [inline src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java]: Not yet; I'll create a JIRA for tracking that.

**Comment by cah292** [comment]
I've committed more changes based on feedback. This is ready for review again.

**Comment by cah292** [comment]
retest this please

**Pull request merged**